### PR TITLE
Add tests for popup widgets and fix perspective popup slider initialization

### LINKED ISSUE #3794: Add tests for popups
## 🧰 Task

I realized after reviewing #3766 , that I did not include tests for some of the popup dialogs I worked on (the grid mode popup and the playback settings popup).
These need to be added...

### PR DESCRIPTION
**Pull request by dalthviz**
# Fixes/Closes

Closes #3794

# Description

This adds some tests to check for popup widgets functionality (popups that are launched from the buttons that handle some viewer functionality and from the dims slider play button):

* dims slider play button -> play settings (fps, play mode, loop mode) popup
* roll dims button -> dims order/sorter popup
* grid button -> grid settings (grid stride, width and height) popup
* ndisplay button -> perspective slider popup

While checking this, I noticed an issue with the perspective popup that can be launched from the ndisplay button:

<details>

```python-traceback
Traceback (most recent call last):
  File "E:\Acer\Documentos\Quansight\Napari\napari\napari\_qt\widgets\qt_viewer_buttons.py", line 186, in open_perspective_popup
    sld.setValue(self.viewer.camera.perspective)
TypeError: setValue(self, a0: int): argument 1 has unexpected type 'float'
```

</details>

So I added in this PR the missing cast to `int` so the related new test can pass.

## Notes: 

Checking the logic the popups creation follows in the `QtViewerButtons` class, maybe could be worthy to move the detailed widget creation for popups (like the perspective and grid settings popups) to be in its own class definition? So, change all the popups that are created from the viewer buttons to follow a similar approach like the one the dims sorter popup uses where a `QtDimsSorter` class is defined?

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue) - add tests

# How has this been tested?

- [x] all tests pass with my change
- [x] I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [x] My PR is the minimum possible work for the desired functionality
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: napari/_qt/widgets/qt_viewer_buttons.py [hunk 1]
```diff
@@ -183,10 +183,11 @@ def open_perspective_popup(self):
         # make slider connected to perspective parameter
         sld = QSlider(Qt.Orientation.Horizontal, self)
         sld.setRange(0, max(90, int(self.viewer.camera.perspective)))
-        sld.setValue(self.viewer.camera.perspective)
+        sld.setValue(int(self.viewer.camera.perspective))
         sld.valueChanged.connect(
             lambda v: setattr(self.viewer.camera, 'perspective', v)
         )
+        self.perspective_slider = sld
```
Comment (psobolewskiPhD) [inline napari/_qt/widgets/qt_viewer_buttons.py:190]: This is for the tests right?

**Comment by psobolewskiPhD** [comment]
> * ndisplay button -> perspective slider popup
> 
> While checking this, I noticed an issue with the perspective popup that can be launched from the ndisplay button:

🤯 
Didn't even know there was such a popup!
It's not documented in [the viewer guide](https://napari.org/dev/tutorials/fundamentals/viewer.html#viewer-buttons)
Turns out it's not new either: https://github.com/napari/napari/pull/2636

Anyhow, I can confirm the error and that the fix works:
image

Pretty neat effect—not sure how useful...

Researching the popup I learned you can shift-drag to move the 3D rendering! This also isn't documented but is super useful. 
I will make a quick docs PR for both of these that depends on this one being merged.

**Comment by brisvag** [comment]
Note that vispy by default allows shift+right-click-drag to change fov as well :P

**Comment by psobolewskiPhD** [comment]
> Note that vispy by default allows shift+right-click-drag to change fov as well :P

Cool! I didn't know that. I'll add to the docs PR.
Edit: https://github.com/napari/docs/pull/159/commits/7c70b2303fad3f5889e38d1def26072d15b411ab

File: napari/_qt/widgets/qt_viewer_buttons.py (hunk 1 above)
Comment (dalthviz) [reply napari/_qt/widgets/qt_viewer_buttons.py:190]: Yes! This enables doing the test in a simpler manner. However, I also added this here to keep some consistency with the way the grid mode popup is implemented (there also a reference to the widgets from the popup is created):

https://github.com/napari/napari/blob/87108ea1cb11501f5659b9d97a80c27a61a6ce32/napari/_qt/widgets/qt_viewer_buttons.py#L238-L272

File: napari/_qt/widgets/_tests/test_qt_viewer_buttons.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    """
+    Make sure the QtViewerButtons.gridViewbutton popup works.
+
+    The popup widget should be able to show/change viewer grid settings.
+    """
+    # get viewer model and buttons
+    viewer, viewer_buttons = qt_viewer_buttons
+    assert viewer_buttons.gridViewButton
+
+    # make grip settings popup
```
Comment (psobolewskiPhD) [inline napari/_qt/widgets/_tests/test_qt_viewer_buttons.py]:
```suggestion
    # make grid settings popup
```

**Review by psobolewskiPhD — APPROVED** [review]
Thanks @dalthviz !
regarding:
> Checking the logic the popups creation follows in the QtViewerButtons class, maybe could be worthy to move the detailed widget creation for popups (like the perspective and grid settings popups) to be in its own class definition? So, change all the popups that are created from the viewer buttons to follow a similar approach like the one the dims sorter popup uses where a QtDimsSorter class is defined?

I think it may be best to open an issue for better visibility?

**Comment by dalthviz** [comment]
Note: Seems like the addition of tests is now making jobs to be cancelled due to they needing more time to finish? Maybe increasing the `timeout-minutes` value for the `test` job and `Test with tox` step in 5-10 mins?

**Pull request merged**